text/0000-ubi-based-stacks.md

```diff
@@ -0,0 +1,181 @@
+buildpacks perform the download & install of depdencies within layers, the preferred route would be to to use the 
+dependencies supplied via the configured package repositories within the images. Eg, rather than downloading a 
+JVM/JDK, one would be installed via 'yum' or 'dnf'. 
+
+Until recently, CNCF Buildpacks were unable to effect changes to the Builder or Run image at a system level (eg, you
+could not make modifications that required root from a buildpack, such as installing packages via yum). Recent 
+updates to the Buildpacks project have added 'extensions', these can be considered as special variants of buildpacks 
+that are able to modify the Builder image (and substitute the Run image), via generated Dockerfiles created during 
+the execution of the Extensions. (In the future, the goal is to also allow modification of the Run image via the 
+same mechanism).
+
+Via extensions, it becomes possible to have a Builder image that can install dependencies such as Java, Maven,
+NodeJS, etc via yum. These extensions are very similar to buildpacks, participating in the 'detect' phase, and 
+being allowed to issue 'provides' to the buildplan (but critically not 'requires'). Then extensions that are
+selected to become part of a build run a 'generate' task (rather than a buildpacks 'build' task), that may output 
+Dockerfiles to modify the Builder image, and/or substitute the Run image. Just like buildpacks, the detect & 
+generate invocations get access to the source project to determine what actions they wish to take. For more 
+information on extensions see.. https://github.com/buildpacks/spec/blob/main/image_extension.md
+
+### Initial offered image
+
+The intent is provide an initial builder image that caters to Java & NodeJS runtimes (without Java Native). 
+Some early prototyping has shown this to be possible using the existing Java & NodeJS builder ecosystems 
+present within Paketo today, with Extensions installing the main runtime dependencies. 
+
+In the case of Java, much work has already been done by Paketo to allow for different buildpacks to supply the 
+JDK/JVM, making it relatively simple to extend this to allow an Extension to perform the same role. For NodeJS, there
+were some small changes required, that have already been merged. 
+
+The initial image will be labelled with a Stack ID of `io.buildpacks.stacks.ubi8` Note that much as Ubuntu
+moves from release to release, so does UBI, and currently we will target UBI8, but want to ensure the naming allows
+for the future addition for UBI9 and so on.
+
+There will be no "intermediate" image as with Bionic, as the base builder image is intended to be a published
+UBI8 image.
+
+The initial image shall be considered as the 'base' variant, for the purposes of differentiation when/if 
+subsequent images are published. 
+
+### Image Naming and Tagging
+
+The stack will name and tag the release images with the following pattern:
```

Comment:
I don't think it would necessarily impact the naming of stacks, but is it a goal to have multi-architecture images available? Maybe put another way, will these stacks support both AMD64 and ARM64?

Reply:
I think multi-arch makes sense, I don't specifically see any complications with that from a ubi perspective, provided ubi images exist for the arch's being considered, because the deps are installed via rpm, so that's just a shell interaction, the appropriate rpm repo for the arch would already be configured in the base image, so I think if there any multi arch concerns for Paketo UBI stacks, they are existing ones shared by Paketo already (eg, the Go detect/build binaries needing to be executable etc).

Reply:
Added note =)